06_gpu_and_ml/yolo/yolo.py

```diff
@@ -0,0 +1,288 @@
+# ---
+# runtimes: ["runc", "gvisor"]
+# ---
+#
+# # Training and Inferencing YOLOv10
+#
+# Example by [@AnirudhRahul](https://github.com/AnirudhRahul)
+#
+# [YOLOv10](https://docs.ultralytics.com/models/yolov10/) is the most modern version of the popular "You Only Look Once" ("YOLO") computer vision model, 
+# which has been trained to perform bounding box object detection.
+#
+# In this example, we use the V10 model via the [Ultralytics package](https://docs.ultralytics.com/). We will:
+# - Fine tune on a custom dataset, in our case the DoclayNet dataset for object detection within documents.
+# - Deploy an inference endpoint.
+
+# For commercial use, sure to consult the [ultralytics software license](https://docs.ultralytics.com/#yolo-licenses-how-is-ultralytics-yolo-licensed).
+
+# ## Building Our Image
+#
+# The example uses the Ultralytics package to run training inference and the Huggingface dataset package to load DoclayNet data.
+# 
+# All packages are installed into a Pytorch 2.3.1 base image
+# using the `pip_install` function.
+#
+# In addition, we download the base YOLO model weights into the container at build-time
+# using the `run_function` function
+
+import modal
+import urllib.request, os, yaml, json, shutil, zipfile
+from pathlib import Path
+
+def load_yolo_weights():
+    from ultralytics import YOLO
+    # Choose the large size checkpoint
```

Comment:
Any comments on why this size?

For example: even if we turn this into a webcam demo using websockets, I don't think we actually care about the extra few ms of latency from XL/L vs S/N. Good to call out latency numbers, so folks know how to adjust this example.